### ISSUE TITLE
Support for slf4j-ext

### ISSUE THREAD

**Issue by corneil** [MicroUtils/kotlin-logging#71]
How about support for slf4j-ext [https://www.slf4j.org/extensions.html#extended_logger](https://www.slf4j.org/extensions.html#extended_logger)

This would imply something like:
```
actual interface KXLogger : XLogger, KLogger {
 // similar to KLogger with added methods for XLogger
}

open class KXLogging : KXLoggable {
    override val logger: KXLogger = logger()
}
```

I would be up for the PR if you are interested.

**Comment by oshai** [comment]
Sure, a PR is welcome!

**Comment by corneil** [comment]
Do you think `KotlinLogging` and `KLoggerFactory` should be expanded with methods for the `XLogger` or should we do new classes?
slf4j-ext is a relatively small dependency that extends the slf4j `Logger`
The JS implementation can just add the methods and leverage off the existing JS implementation to add `entry`, `exit`, `throwing` etc.

**Comment by corneil** [comment]
Should we use `KXLogger` or `XKLogger`?

**Comment by oshai** [comment]
I would call it KXLogger. I'm not sure why those methods in slf4j are not in Logger interface. Is that compatibility issue? Probably changing hierarchy is not the best idea. If you have another idea please explain it.

**Comment by corneil** [comment]
XLogger extends Logger and adds methods.
So I think KXLogger should extends KLogger and XLogger.
Thus like with slf4j you have normal Logger and XLogger, in this case, KLogger and KXLogger.
I will make new factories where the XLogger or KXLogger enters the picture.
The only challenge I see is that enter and exit in slf4j report the calling method name but if wrapped the caller is KXLogger.enter or KXLogger.exit. I will dig a bit to find how to solve this.

**Comment by oshai** [comment]
I will continue the discussion on the PR: https://github.com/MicroUtils/kotlin-logging/pull/72

Closing the issue here.

**Comment by oshai** [comment]
fixed in #73
